### PR TITLE
fix(migrations): properly bundle shared compiler code into migrations

### PR DESCRIPTION
**Pull request by devversion**
See individual commits

Related to #58455

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -161,7 +161,8 @@
     "@angular-devkit/architect-cli": "0.1900.0-rc.0",
     "@angular/animations": "^19.0.0-next",
     "@angular/build-tooling": "https://github.com/angular/dev-infra-private-build-tooling-builds.git#66f78f64d9ab33ae430c527d3038b8c59b2d9a7f",
-    "@angular/core": "^19.0.0-next",
+    "@angular/common": "^19.0.0-rc.0",
```
Comment (crisbeto) [inline package.json:164]: Why do we need `common` here?

Comment (devversion) [reply package.json:164]: oops. That's a mistake/leftover from my `adev` testing. good catch

**Review by crisbeto — APPROVED** [review]

**Comment by devversion** [comment]
[TGP](https://fusion2.corp.google.com/presubmit/693380649/OCL:693380649:BASE:693330118:1730841441591:706a8c2d/targets)

**Comment by pkozlowski-opensource** [comment]
This PR was merged into the repository by commit 121b340916f0bda54cb3b668ddfa99cb9b7a3d6d.

The changes were merged into the following branches: main, 19.0.x

**Pull request closed without merging**